doc/languages-frameworks/python.section.md

```diff
@@ -1070,8 +1076,15 @@ stdenv.mkDerivation {
   shellHook = ''
     # set SOURCE_DATE_EPOCH so that we can use python wheels
     SOURCE_DATE_EPOCH=$(date +%s)
-    virtualenv --python=${pythonPackages.python.interpreter} --no-setuptools venv
-    export PATH=$PWD/venv/bin:$PATH
+
+    if [ -d "${venvDir}" ]; then
+      echo "Skipping venv creation, '${venvDir}' already exists"
+    else
+      echo "Creating new venv environment in path: '${venvDir}'"
+      ${pythonPackages.python.interpreter} -m venv "${venvDir}"
+    fi
+
+    source "${venvDir}/bin/activate"
```

Comment:
you'll probably want to add something like this if you want the packages installed by venv to be picked up by the interpreter:
```suggestion
    source "${venvDir}/bin/activate"
    # add venv dir to python search path for modules
    export PYTHONPATH=`pwd`/$venvDir/${python.sitePackages}/:$PYTHONPATH
```

Reply:
For me this is not necessary and it picks up locally installed libraries correctly. Could you double-check?

Reply:
yep, just checked, not needed.

May have been fixed since i started doing this a year ago